code/_onclick/hud/plane_master.dm

```diff
@@ -5,21 +5,29 @@
 	blend_mode = BLEND_OVERLAY
 	var/show_alpha = 255
 	var/hide_alpha = 0
+	var/current_filters = list()
 
 /obj/screen/plane_master/proc/Show(override)
 	alpha = override || show_alpha
 
 /obj/screen/plane_master/proc/Hide(override)
 	alpha = override || hide_alpha
 
-/obj/screen/plane_master/proc/outline(_size, _color)
-	filters += filter(type = "outline", size = _size, color = _color)
+/obj/screen/plane_master/proc/add_filter(key, filter)
+	if(current_filters[key])
+		filters -= current_filters[key]
 
-/obj/screen/plane_master/proc/shadow(_size, _border, _offset = 0, _x = 0, _y = 0, _color = "#04080FAA")
-	filters += filter(type = "drop_shadow", x = _x, y = _y, color = _color, size = _size, offset = _offset)
+	filters += filter
+	current_filters[key] = filters[filters.len] //assigning `filter` will not allow it to work later while animating removal
```

Comment:
When I try to do something like

```var/filter/filter_to_remove = current_filters[key] //ref saved before the filter was applied
animate(filter_to_remove, ...)
```

BYOND will throw an exception saying that "abstract" filters cannot be animated, which is extremely odd. It somehow pulls information on what object this filter belongs to from an access like `filters[filters.len]`, as in the BYOND docs, but what I do here also works, so I have no clue how this is implemented. Magic.